File: simple.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
	"fmt"
)

func main() {
	foo := hmac.New(sha256.New, []byte("secret"))
	foo.Write([]byte("first-value"))
	fooDigest := foo.Sum(nil)

	// Convert []byte to hex string using Sprintf
	fooHex := fmt.Sprintf("%x", fooDigest)
	fmt.Printf("foo hex: %s\n", fooHex)

	// Convert hex string back to []byte
	fooBytes, err := hex.DecodeString(fooHex)
	if err != nil {
		panic(err)
	}

	// Use the converted bytes
	bar := hmac.New(sha256.New, fooBytes)
	bar.Write([]byte("second-value"))
	barDigest := bar.Sum(nil)
	fmt.Printf("bar: %x\n", barDigest)
}

File: simple.ts
import crypto from 'crypto'

const foo = crypto.createHmac('sha256', 'secret').update('first-value').digest()
console.log(`foo`, foo.toString('hex'))
const bar = crypto.createHmac('sha256', foo).update('second-value').digest()
console.log(`bar`, bar.toString('hex'))
{
  const bar = crypto
    .createHmac('sha256', foo.toString())
    .update('second-value')
    .digest()
  console.log(`bar`, bar.toString('hex'))
}
